### PR TITLE
Simplify zephyr entropy

### PR DESCRIPTION
**Pull request by SebastianBoe**
Simplify zephyr entropy.

See commit messages for details.

### CODE REVIEW

**Review by mswarowsky — APPROVED** [review]
nice catch

**Comment by joerchan** [comment]
Nice simplification to avoid the potential circular calling

File: subsys/nrf_security/src/drivers/zephyr/Kconfig [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-config PSA_NEED_ZEPHYR_ENTROPY_DRIVER
+config PSA_NEED_NRF_RNG_ENTROPY_DRIVER
```
[the remark this answers is not in the document]
Comment (SebastianBoe) [reply subsys/nrf_security/src/drivers/zephyr/Kconfig:7]: Good thinking :)

[the remark this answers is not in the document]
Comment (SebastianBoe) [reply subsys/nrf_security/src/drivers/zephyr/Kconfig:7]: Done

**Review by joerchan — APPROVED** [review]

**Pull request merged**